services/payments/src/simcore_service_payments/services/payments.py

```diff
@@ -206,23 +213,41 @@ async def init_payment_with_payment_method(
         ),
     )
 
-    payment_id = await repo_transactions.insert_init_payment_transaction(
-        payment_id=payment_inited.payment_id,
-        price_dollars=amount_dollars,
-        osparc_credits=target_credits,
-        product_name=product_name,
-        user_id=user_id,
-        user_email=user_email,
-        wallet_id=wallet_id,
-        comment=comment,
-        initiated_at=initiated_at,
-    )
+    payment_id = ack.payment_id
 
-    return WalletPaymentInitiated(
-        payment_id=f"{payment_id}",
-        payment_form_url=None,
+    async for attempt in AsyncRetrying(
+        stop=stop_after_attempt(3),
```

Comment:
Suggestion: Since you are dealing with an external system that can be offline due to load, I think `wait_random_exponential` would also help here.

Reply:
but this is just retrying  to ensure it has different uuids.
 if the system gets offline, i do not want it to retry